### ISSUE TITLE
Text in prompt on wrong position when activity smaller then screen and not centered

### ISSUE THREAD

**Issue by ElineDeMeyer** [sjwall/MaterialTapTargetPrompt#17]
When you have an activity like in your DialogActivity but it is not in the center of the screen (eg at the bottom) the prompt is shown ok but the text in it is positioned wrong.

In my example the targetView is on the top right of my activity but my activity is at bottom center of the window. The targetView is still in the bottom right part of the window.
Parameters tell met the text will be shown above/left of the target but that is not in my activity anymore.

I'm searching for a solution in order to help you out :)

**Comment by sjwall** [comment]
Hi, it sounds like the prompt should be checking if the view has enough height to display the text above (or below) the target and if it doesn't then display the text it to the side. The other option is to add functionality to disable clipping to the activity bounds so that it could draw outside the activity bounds. 

Let me know how you get on searching for a solution. I shall look at expanding the dialog functionality in the sample app to cover having the dialog in different positions and sizes.

**Comment by ElineDeMeyer** [comment]
Its ok to draw the prompt inside the bounds. I'm perfectly happy with the prompt. It is only shown in the activity bounds. Just the text is not there. 

I think if he knew the parent is not full screen parent decorview but is a coordinatorlayout smaller then the screen, it should be fine.

**Comment by ElineDeMeyer** [comment]
Hi Samuel,

I checked your library and tried fixing it so that for activities smaller
than full screen the parent is the activity and not de decor view. Although
al my values seem to be right at the moment, I'm still not able to solve
this. Can you please check how this can be done?

Regards
Eline

2016-10-19 14:13 GMT+02:00 Samuel Wall <notifications@github.com>:

**Comment by ElineDeMeyer** [comment]
Changes I did:
* in getParentView() where you check if the contentView is a drawer layout, I added comparing the size of the contentView with the one of the decorView in that way we pass the contentView when the activity is not full screen:
```
if (contentView.getClass().getName().equals("android.support.v4.widget.DrawerLayout")
                    || contentView.getHeight() != decorView.getHeight()
                    || contentView.getWidth() != decorView.getWidth())
```
* in updateTextPositioning() when calculating mView.mPrimaryTextTop I added in both cases the getParentView().getTop(). This will be 0 for full screen activity and otherwise we get the position of the text relative to the screen boundaries
```
mView.mPrimaryTextTop = mView.mCentreTop;
        if (mTextPositionAbove)
        {
            mView.mPrimaryTextTop = mView.mPrimaryTextTop - mBaseFocalRadius - mFocalToTextPadding - mView.mPrimaryTextLayout.getHeight() + getParentView().getTop();
        }
        else
        {
            mView.mPrimaryTextTop += mBaseFocalRadius + mFocalToTextPadding + getParentView().getTop();
        }
```

1 problem left: 
after calculating everything and ready to show, I get an onGlobalLayout event and updateFocalCentrePosition is called again but now it sais that my FAB is top right of my full screen instead of top right of my dialogActivity 
to test I added android:layout_gravity="center_horizontal|bottom" to the CoordinatorLayout and android:layout_gravity="bottom|end" to the FAB in the activity_dialog_style.xml.

Any idea?

**Comment by sjwall** [comment]
Hey, thanks for your work. I shall have a look at it and see what I can do to fix it

**Comment by ElineDeMeyer** [comment]
Debugged a little further and I saw that the mClipToView when having the onGlobalLayout has a different left and top (left: 64 iso 510 and top: 64 iso 1536). It's then used to update everything and the activity is now larger and almost full screen. mClipToView is still the same mClipToView as before (according to my watches in Android Studio). No idea why it changed yet...

**Comment by sjwall** [comment]
Still need to do some more testing but changing the text positioning at the end of `updateFocalCentrePosition()` fixes it for me:

```Java
mTextPositionAbove = mView.mCentreTop > mView.mClipBoundsTop + ((mView.mClipBoundsBottom - mView.mClipBoundsTop) / 2);
mTextPositionRight = mView.mCentreLeft > mView.mClipBoundsLeft + ((mView.mClipBoundsRight - mView.mClipBoundsLeft) / 2);
```

I think this can (and should) be improved further so that it can position the text better when there is limited room

**Comment by ElineDeMeyer** [comment]
So only this without my changes?

**Comment by sjwall** [comment]
Currently, yes

**Comment by sjwall** [comment]
@ElineDeMeyer did the changes that I suggested work for you?

**Comment by ElineDeMeyer** [comment]
Yes, please include these or an improved solution in one of your next releases :)

**Comment by sjwall** [comment]
Thanks for your help, it is very much appreciated! The fix is released in v1.5.0
